drivers/gpio/gpio_max22190.c

```diff
@@ -0,0 +1,658 @@
+	enum max22190_ch_state channels[MAX22190_CHANNELS];
+	enum max22190_ch_wb_state wb[MAX22190_CHANNELS];
+	union max22190_cfg cfg;
+	union max22190_fault1 fault1;
+	union max22190_fault1_en fault1_en;
+	union max22190_fault2 fault2;
+	union max22190_fault2_en fault2_en;
+};
+
+/*
+ * @brief Compute the CRC5 value for MAX22190
+ * @param data - Data array to calculate CRC for.
+ * @return CRC result.
+ */
+static uint8_t max22190_crc(uint8_t *data)
+{
+	int length = 19;
+	uint8_t crc_step, tmp;
+	uint8_t crc_init = 0x7;
+	uint8_t crc_poly = 0x35;
+	int i;
+
+	/*
+	 * This is the C custom implementation of CRC function for MAX22190, and
+	 * can be found here:
+	 * https://www.analog.com/en/design-notes/guidelines-to-implement-crc-algorithm.html
+	 */
+	uint32_t datainput = (uint32_t)((data[0] << 16) + (data[1] << 8) + data[2]);
+
+	datainput = (datainput & 0xFFFFE0) + crc_init;
+
+	tmp = (uint8_t)((datainput & 0xFC0000) >> 18);
+
+	if ((tmp & 0x20) == 0x20) {
+		crc_step = (uint8_t)(tmp ^ crc_poly);
+	} else {
+		crc_step = tmp;
+	}
+
+	for (i = 0; i < length - 1; i++) {
+		tmp = (uint8_t)(((crc_step & 0x1F) << 1) +
+				((datainput >> (length - 2 - i)) & 0x01));
+
+		if ((tmp & 0x20) == 0x20) {
+			crc_step = (uint8_t)(tmp ^ crc_poly);
+		} else {
+			crc_step = tmp;
+		}
+	}
+
+	return (uint8_t)(crc_step & 0x1F);
+}
+
+/*
+ * @brief Update chan WB state in max22190_data
+ *
+ * @param dev - MAX22190 device.
+ * @param val - value to be set.
+ */
+static void max22190_update_wb_stat(const struct device *dev, uint8_t val)
+{
+	struct max22190_data *data = dev->data;
+
+	for (int ch_n = 0; ch_n < 8; ch_n++) {
+		data->wb[ch_n] = (val >> ch_n) & 0x1;
+	}
+}
+
+/*
+ * @brief Update chan IN state in max22190_data
+ *
+ * @param dev - MAX22190 device.
+ * @param val - value to be set.
+ */
+static void max22190_update_in_stat(const struct device *dev, uint8_t val)
+{
+	struct max22190_data *data = dev->data;
+
+	for (int ch_n = 0; ch_n < 8; ch_n++)	{
+		data->channels[ch_n] = (val >> ch_n) & 0x1;
+	}
+}
+
+/*
+ * @brief Register write function for MAX22190
+ *
+ * @param dev - MAX22190 device config.
+ * @param addr - Register value to which data is written.
+ * @param val - Value which is to be written to requested register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int max22190_reg_transceive(const struct device *dev, uint8_t addr, uint8_t val, uint8_t rw)
+{
+	uint8_t crc;
+	int ret;
+
+	uint8_t local_rx_buff[MAX22190_MAX_PKT_SIZE] = {0};
+	uint8_t local_tx_buff[MAX22190_MAX_PKT_SIZE] = {0};
+
+	const struct max22190_config *config = dev->config;
+
+	struct spi_buf tx_buf = {
+				.buf = &local_tx_buff,
+				.len = config->pkt_size,
+				};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+
+	struct spi_buf rx_buf = {
+				.buf = &local_rx_buff,
+				.len = config->pkt_size,
+				};
+	const struct spi_buf_set rx = {.buffers = &rx_buf, .count = 1};
+
+	if (config->crc_en & 0) {
```

Comment:
Thanks! This exact is not needed because in current implementation length vary due to `crc_en`. 

`pkt_size = !(DT_INST_PROP(id, max22190_mode) & 0x1) ? 3 : 2`